### PR TITLE
Adds getNextFederationCreationBlockHeight tests to FederationStorageP…

### PR DESCRIPTION
**Pull request by jeremy-then**
Adds getNextFederationCreationBlockHeight tests to FederationStorageProviderImplTests and removes the ones from BridgeStorageProviderTest.

### CODE REVIEW

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 1]
```diff
@@ -806,11 +806,11 @@ void getFederationElection_whenElectionIsNotInStorage_shouldReturnDefault() {
     @Test
     void getActiveFederationCreationBlockHeight_beforeRSKIP186_storageIsNotAccessedAndReturnsEmpty() {
 
+        // Arrange
+
         ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
         when(activations.isActive(ConsensusRule.RSKIP186)).thenReturn(false);
 
-        // Arrange
-
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:813]: i think the activations are part of the arrange

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 2]
```diff
@@ -901,6 +903,82 @@ void setActiveFederationCreationBlockHeight() {
...
+    void getNextFederationCreationBlockHeight_beforeRSKIP186_storageIsNotAccessedAndReturnsEmpty() {
+
+        // Arrange
+
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP186)).thenReturn(false);
+
+        StorageAccessor storageAccessor = new InMemoryStorage();
+        // Putting some value in the storage just to then assert that before fork, the storage won't be accessed.
+        storageAccessor.saveToRepository(NEXT_FEDERATION_CREATION_BLOCK_HEIGHT_KEY.getKey(), new byte[] { 1 });
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:916]: i think what you did [here](https://github.com/rsksmart/rskj/pull/2544/files#diff-f42b277f39730e846e15f899b88e1e7e718280d3ea3d4c4d5b035a5d66385923R837) is better :)

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 3]
```diff
@@ -901,6 +903,82 @@ void setActiveFederationCreationBlockHeight() {
...
+    void getNextFederationCreationBlockHeight_afterRSKIP186_getsValueFromStorage() {
+
+        // Arrange
+
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP186)).thenReturn(true);
+
+        StorageAccessor storageAccessor = new InMemoryStorage();
+        long expectedValue = 1;
+        storageAccessor.saveToRepository(NEXT_FEDERATION_CREATION_BLOCK_HEIGHT_KEY.getKey(), new byte[] { 1 });
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:937]: same here

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 4]
```diff
@@ -901,6 +903,82 @@ void setActiveFederationCreationBlockHeight() {
...
+        Optional<Long> actualValueOptional = federationStorageProvider.getNextFederationCreationBlockHeight(activations);
+
+        // Assert
+
+        assertTrue(actualValueOptional.isPresent());
+        assertEquals(expectedValue, actualValueOptional.get());
+
+        // Setting in storage a different value to assert that calling the method again should return cached value
+
+        storageAccessor.saveToRepository(NEXT_FEDERATION_CREATION_BLOCK_HEIGHT_KEY.getKey(), new byte[] { 2 });
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:951]: and here

**Review by apancorb — APPROVED** [review]
LGTM

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java (hunk 2 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:916]: In this test, the `new byte[] { 1 }` is not an `"expected"` value. It won't use anywhere. No need to put in into a variable.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java (hunk 3 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:937]: It makes sense here.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java (hunk 4 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:951]: Not here.

**Review by marcos-iov — COMMENTED** [review]
Same comment as in other PRs regarding values used and activation setup. Other than that looks good

**Pull request closed without merging**